src/status_im2/contexts/wallet/send/events.cljs

```diff
@@ -0,0 +1,78 @@
+(ns status-im2.contexts.wallet.send.events
+  (:require
+    [status-im2.constants :as constants]
+    [taoensso.timbre :as log]
+    [utils.datetime :as datetime]
+    [utils.money :as money]
+    [utils.number]
+    [utils.re-frame :as rf]))
+
+(rf/reg-event-fx :wallet/suggested-routes-success
+ (fn [{:keys [db]} [suggested-routes timestamp]]
+   (when (= (get-in db [:wallet :ui :send :suggested-routes-call-timestamp]) timestamp)
+     {:db (-> db
+              (assoc-in [:wallet :ui :send :suggested-routes] suggested-routes)
+              (assoc-in [:wallet :ui :send :route] (first (:Best suggested-routes)))
+              (assoc-in [:wallet :ui :send :loading-suggested-routes?] false))})))
+
+(rf/reg-event-fx :wallet/suggested-routes-error
+ (fn [{:keys [db]} [_error]]
+   {:db (-> db
+            (update-in [:wallet :ui :send] dissoc :suggested-routes)
+            (update-in [:wallet :ui :send] dissoc :route)
+            (assoc-in [:wallet :ui :send :loading-suggested-routes?] false))}))
+
+(rf/reg-event-fx :wallet/clean-suggested-routes
+ (fn [{:keys [db]}]
+   {:db (-> db
+            (update-in [:wallet :ui :send] dissoc :suggested-routes)
+            (update-in [:wallet :ui :send] dissoc :route)
+            (update-in [:wallet :ui :send] dissoc :loading-suggested-routes?))}))
+
+(rf/reg-event-fx :wallet/select-send-address
+ (fn [{:keys [db]} [address]]
+   {:db (assoc-in db [:wallet :ui :send :to-address] address)}))
+
+(rf/reg-event-fx :wallet/send-select-token
+ (fn [{:keys [db]} [token stack-id]]
+   {:db (assoc-in db [:wallet :ui :send :token] token)
+    :fx [[:navigate-to-within-stack [:wallet-send-input-amount stack-id]]]}))
+
+(rf/reg-event-fx :wallet/get-suggested-routes
+ (fn [{:keys [db]} [amount]]
+   (let [wallet-address      (get-in db [:wallet :current-viewing-account-address])
+         token               (get-in db [:wallet :ui :send :token])
+         to-address          (get-in db [:wallet :ui :send :to-address])
+         token-decimal       (:decimals token)
+         token-id            (:symbol token)
+         network-preferences [constants/mainnet-chain-id] ; TODO: don't hardcode network preferences
```

Comment:
can we remove this? the network preferences are currently available and by default will be all valid networks anyway 👍

Reply:
Correct me if I'm wrong, but this value should be based on network preferences from the receiver? Not sure